src/main/java/radiounit/DemoOneRadioUnit.java

```diff
@@ -1,154 +1,170 @@
-	}
-
-	@Override
-	public String getRadioUnitName() {
-		System.out.println("Get Radio Unit Name");
-		return name;
-	}
-
-	@Override
-	public String getVendor() {
-		System.out.println("Get Vendor");
-		return vendor;
-	}
-
-	@Override
-	public RatType getRatType() {
-		System.out.println("Get Rat Type");
-		return ruType;
-	}
-
-	@Override
-	public List<Carrier> getCarriers() {
-		System.out.println("Get Carriers");
-		return carriers;
-	}
-
-	@Override
-	public AlarmStatusLevel getAlarmStatus() {
-		System.out.println("Get Alarm Status");
-		return alarm;
-	}
-
-	@Override
-	public void setRadioUnitName(String name) {
-		System.out.println("Set Radio Unit Name");
-		this.name = name;
-	}
-
-	@Override
-	public void raiseAlarm(AlarmStatusLevel alarm) {
-		System.out.println("Raise Alarm");
-		this.alarm = alarm;
-	}
+    List<Carrier> carriers;
+    String ipAddress;
+    String name;
+    Vendor vendor;
+    RatType ruType;
+    AlarmStatusLevel alarm;
+
+    public DemoOneRadioUnit(String name, Vendor vendor, RatType ruType) {
+        this.carriers = new ArrayList<>();
+        this.ipAddress = "1.1.1.1";
+        this.name = name;
+        this.vendor = vendor;
+        this.ruType = ruType;
+        this.alarm = AlarmStatusLevel.NO_ALARM;
+    }
+
+    @Override
+    public void setup() {
+        System.out.println("[DEBUG] Setting up radio unit");
+
+    }
+
+    @Override
+    public void activate() {
+        System.out.println("[DEBUG] Activating radio unit");
+
+    }
+
+    @Override
+    public void deactivate() {
+        System.out.println("[DEBUG] Deactivating radio unit");
+
+    }
+
+    @Override
+    public void release() {
+        System.out.println("[DEBUG] Releasing");
+
+    }
+
+    @Override
+    public void signalScaling() {
+        System.out.println("[DEBUG] Signal scaling");
+
+    }
+
+    @Override
+    public void postActivation() {
+        System.out.println("[DEBUG] Post activation");
+
+    }
+
+    @Override
+    public void performSelfDiagnostics() {
+        System.out.println("[DEBUG] performSelfDiagnostics");
+
+    }
+
+    @Override
+    public void acknowledgeAlarm() {
+        System.out.println("[DEBUG] Acknowledge alarm");
+
+    }
+
+    @Override
+    public void setupCarrier(Carrier carrier) {
+        System.out.println("[DEBUG] Setting up carrier");
+        carriers.add(carrier);
+
+    }
+
+    @Override
+    public void modifyCarrier(int carrierId, FrequencyBand freq) {
+        System.out.println("[DEBUG] Modifying carrier");
+        carriers.forEach(carrier -> {
+            if (carrier.getCarrierId() == carrierId) {
+                //carrier.setFrequencyBand(freq);
+                return;
+            }
+        });
+    }
+
+    @Override
+    public void removeCarrier(int carrierId) {
+        System.out.println("[DEBUG] Remove carrier");
+        carriers.forEach(carrier -> {
+            if (carrier.getCarrierId() == carrierId) {
+                //carriers.remove(carrier);
```

Comment:
We should uncomment this

Reply:
The comments can be removed, I left it like this since we didn't actually make use of the functionality in demo 1 at all, since it wasn't a part of the requirements we were given. These were just some functions I threw together quickly while trying to have a RU work with the UI.